### PR TITLE
Add missing timezones to zone-index

### PR DESCRIPTION
**Pull request by hellium01**
Added additional time zones that are included until IANA time zone data version 2017a.

### CODE REVIEW

File: presto-spi/src/test/java/com/facebook/presto/spi/type/TestTimeZoneKey.java [hunk 1]
```diff
@@ -192,6 +192,6 @@ public int compare(TimeZoneKey left, TimeZoneKey right)
             hasher.putString(timeZoneKey.getId(), StandardCharsets.UTF_8);
         }
         // Zone file should not (normally) be changed, so let's make is more difficult
```
Comment (AlekseiS) [inline presto-spi/src/test/java/com/facebook/presto/spi/type/TestTimeZoneKey.java]: typo: "..let's make IT more ..."

**Comment by haozhun** [comment]
Please also add 2017b zones, which kids 2.9.9 supports.

**Comment by haozhun** [comment]
I would just say "Add missing timezones to zone-index" for commit message

**Comment by hellium01** [comment]
This change should have included all the new timezone added for Joda 2.9.9. No new timezone were added between 20.

File: presto-spi/src/test/java/com/facebook/presto/spi/type/TestTimeZoneKey.java [hunk 2]
```diff
@@ -191,7 +191,7 @@ public int compare(TimeZoneKey left, TimeZoneKey right)
             hasher.putShort(timeZoneKey.getKey());
             hasher.putString(timeZoneKey.getId(), StandardCharsets.UTF_8);
         }
-        // Zone file should not (normally) be changed, so let's make is more difficult
-        assertEquals(hasher.hash().asLong(), 5690324204210439335L, "zone-index.properties file contents changed!");
+        // Zone file should not (normally) be changed, so let's make this more difficult
+        assertEquals(hasher.hash().asLong(), -5839014144088293930L, "zone-index.properties file contents changed!");
```
Comment (findepi) [inline presto-spi/src/test/java/com/facebook/presto/spi/type/TestTimeZoneKey.java]: Does this check give any more confidence about `zone-index.properties` file immutability than review process gives anyway?

Comment (dain) [inline presto-spi/src/test/java/com/facebook/presto/spi/type/TestTimeZoneKey.java]: Absolutely.  You have to touch multiple places and that comment shows up in the diff, which reminds people to not touch that file.

**Pull request merged**